chanzuckerberg/sci-components
# thuang-152398-MenuSelect

1. Add `MenuSelect` component
2. [Figma](https://www.figma.com/file/ywENA9rU56r2FCaDqUpaB3/Drop-Menu-Tabs-Tooltips-Seg-Control-Loading?node-id=2%3A313)
3. [Clubhouse ticket](https://app.clubhouse.io/sci-design-system/story/152398/complex-filter-spike-on-menuselect-component)
4. The scope is only for supporting single select, multiselect, and optional searchbox
5. Note that MenuItem style is not following SDS specs, and that will be addressed in [this ticket](https://app.clubhouse.io/sci-design-system/story/152396/complex-filter-update-sds-menuitem-to-match-design)

Test:
1. Pull branch `thuang-152398-MenuSelect`
2. Run `yarn && yarn start`
3. Check storybook for `MenuSelect` demos!

PTAL thank you 😄 🙏 

Default: SingleSelect
<img width="335" alt="Screen Shot 2021-08-17 at 4 24 12 PM" src="https://user-images.githubusercontent.com/6309723/129813288-c12f2535-97b0-4b83-af1d-74b2a19fb2a2.png">

SingleSelectWithSearch
<img width="348" alt="Screen Shot 2021-08-17 at 4 24 17 PM" src="https://user-images.githubusercontent.com/6309723/129813291-df3693b5-52c4-4e96-ae90-e41fb0172eac.png">

MultiSelectWithSearch
<img width="357" alt="Screen Shot 2021-08-17 at 4 24 38 PM" src="https://user-images.githubusercontent.com/6309723/129813294-d9fdea82-941c-4c5b-b881-0ac2159ef1e4.png">

![demo](https://user-images.githubusercontent.com/6309723/129813274-4787e346-7762-4f8d-b152-bbc4ee2f3bb0.gif)

UPDATE: Now demo uses deletable chip 🎉 !

<img width="257" alt="Screen Shot 2021-08-18 at 11 34 07 AM" src="https://user-images.githubusercontent.com/6309723/129953218-26964e40-556f-41e0-9316-b5cdda296001.png">

![demo](https://user-images.githubusercontent.com/6309723/129953135-55491ea6-e83a-4185-991d-60904ae8941d.gif)
